runtime/src/main/java/io/quarkiverse/renarde/util/QuteResolvers.java

```diff
@@ -131,6 +130,36 @@ void configureEngine(@Observes EngineBuilder builder) {
     }
 
     void registerTemplateInstanceLocaleAndRenderArgs(@Observes EngineBuilder engineBuilder, I18N i18n, RenderArgs renderArgs) {
+
+        if (ProfileManager.getLaunchMode().isDevOrTest()) {
+            engineBuilder.addTemplateInstanceInitializer(templateInstance -> {
+                if (!Arc.container().requestContext().isActive()) {
+                    return;
+                }
+                RoutingContext routingContext = Arc.container().instance(RoutingContext.class).get();
+                MultiMap headers = routingContext.response().headers();
+                Template template = templateInstance.getTemplate();
+                boolean isFragment = template.getClass().getName().equals("io.quarkus.qute.TemplateImpl$FragmentImpl");
+                String parentTemplateId = "";
+                if (isFragment) {
+                    try {
+                        Field rootField = template.getClass().getEnclosingClass().getDeclaredField("root");
+                        rootField.setAccessible(true);
+                        var sectionNode = rootField.get(template);
+                        Field originField = sectionNode.getClass().getDeclaredField("origin");
+                        originField.setAccessible(true);
+                        TemplateNode.Origin origin = (TemplateNode.Origin) originField.get(sectionNode);
+                        parentTemplateId = origin.getTemplateId() + ".";
```

Comment:
You can also cast the template to `io.quarkus.qute.Template.Fragment` and then access the parent with `getOriginalTemplate()`.